api/src/test/java/marquez/db/StatsTest.java

```diff
@@ -108,4 +118,93 @@ public void testGetStatsForLastDay() {
+
+    // (3) Retrieve last day and last week job metrics.
+    List<IntervalMetric> intervalMetricsDay = DB.lastDayJobMetrics();
+    assertThat(intervalMetricsDay).isNotEmpty();
+
+    Optional<Integer> countDay =
+        intervalMetricsDay.stream().map(IntervalMetric::getCount).reduce(Integer::sum);
+    assertThat(countDay).isPresent();
+    assertThat(countDay.get()).isEqualTo(1);
+
+    List<IntervalMetric> intervalMetricsWeek = DB.lastWeekJobMetrics("UTC");
+    assertThat(intervalMetricsWeek).isNotEmpty();
+
+    Optional<Integer> countWeek =
+        intervalMetricsWeek.stream().map(IntervalMetric::getCount).reduce(Integer::sum);
+    assertThat(countWeek).isPresent();
+    assertThat(countWeek.get()).isGreaterThanOrEqualTo(1);
+  }
+
+  @Test
+  public void testGetStatsForDatasets() {
+    // (1) Insert a new namespace.
+    final NamespaceRow namespaceRow = DB.upsert(newNamespaceRow());
+
+    // (2) Insert a new source.
+    final SourceRow sourceRow = DB.upsert(newSourceRow());
+    DB.upsert(sourceRow);
+
+    // (3) Insert a new dataset.
+    final DatasetRow datasetRow =
+        DB.upsert(
+            newDatasetRowWith(
+                namespaceRow.getUuid(),
+                namespaceRow.getName(),
+                sourceRow.getUuid(),
+                sourceRow.getName()));
+    DB.upsert(datasetRow);
+
+    // (4) Retrieve last day and last week dataset metrics.
+    List<IntervalMetric> intervalMetricsDay = DB.lastDayDatasetMetrics();
+    assertThat(intervalMetricsDay).isNotEmpty();
+
+    Optional<Integer> countDay =
+        intervalMetricsDay.stream().map(IntervalMetric::getCount).reduce(Integer::sum);
+    assertThat(countDay).isPresent();
+    assertThat(countDay.get()).isEqualTo(1);
+
+    List<IntervalMetric> intervalMetricsWeek = DB.lastWeekDatasetMetrics("UTC");
+    assertThat(intervalMetricsWeek).isNotEmpty();
+
+    Optional<Integer> countWeek =
+        intervalMetricsWeek.stream().map(IntervalMetric::getCount).reduce(Integer::sum);
+    assertThat(countWeek).isPresent();
+    assertThat(countWeek.get()).isGreaterThanOrEqualTo(1);
+  }
+
+  @Test
+  public void testGetStatsForSources() {
+
+    // (1) Insert a new source.
+    final SourceRow sourceRow = DB.upsert(newSourceRow());
+    DB.upsert(sourceRow);
+
+    // (2) Retrieve last day source metrics.
+    List<IntervalMetric> intervalMetricsDay = DB.lastDaySourceMetrics();
+    assertThat(intervalMetricsDay).isNotEmpty();
+
+    Optional<Integer> countDay =
+        intervalMetricsDay.stream().map(IntervalMetric::getCount).reduce(Integer::sum);
+    assertThat(countDay).isPresent();
+    assertThat(countDay.get()).isGreaterThanOrEqualTo(1);
+
+    List<IntervalMetric> intervalMetricsWeek = DB.lastWeekSourceMetrics("UTC");
+
+    Optional<Integer> countWeek =
+        intervalMetricsWeek.stream().map(IntervalMetric::getCount).reduce(Integer::sum);
+    assertThat(countWeek).isPresent();
+    assertThat(countWeek.get()).isGreaterThanOrEqualTo(1);
```

Comment:
Is there a way to assert the weekly count we expect vs just ensuring it's `>=1`?

Reply:
The counts are attributed to different days depending on which time of day the test is run which is the primary challenge on this one.

Reply:
Ok, cleanup is in now.